client/components/ArtworkForm.js

```diff
@@ -0,0 +1,215 @@
+import React, { Component } from 'react'
+import { Redirect } from 'react-router-dom'
+import { connect } from 'react-redux'
+import { addArtwork, editArtwork, getArtwork } from '../redux/artwork'
+import { mediums } from '../../server/constants'
+
+class ArtworkForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      redirect: false,
+      title: '',
+      description: '',
+      year: '',
+      price: '',
+      quantity: '',
+      //genres: [],
+      medium: '',
+      artistId: '',
+    }
+    this.handleChange = this.handleChange.bind(this)
+    //this.handleGenreChange = this.handleGenreChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
+  }
+
+  componentDidMount() {
+    if (!this.props.artwork && this.props.match.params.id) {
+      this.props.getArtwork(this.props.match.params.id)
+    }
+    if (this.props.isEditing) {
```

Comment:
When I land on the edit page for a newly created artwork I get this error in the console:

Warning: `value` prop on `select` should not be null. Consider using an empty string to clear the component or `undefined` for uncontrolled components.

I think it's because you're setting the artistId/medium to null here. Perhaps add in a check to see if it comes back as null, then set the state to an empty string.

Reply:
i wasn't getting this error when i tried most recently, but i did get it at some point so i tried to do what i thought would fix it? might need to revisit down the line to see if there's a specific case